src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -499,26 +497,13 @@ MulticopterPositionControl::run()
 {
 	hrt_abstime time_stamp_last_loop = hrt_absolute_time(); // time stamp of last loop iteration
 
-	_local_pos_sub = orb_subscribe(ORB_ID(vehicle_local_position));
-	orb_set_interval(_local_pos_sub, 20); // 50 Hz updates
-
 	// get initial values for all parameters and subscribtions
 	parameters_update(true);
 	poll_subscriptions();
 
-	// setup file descriptor to poll the local position as loop wakeup source
-	px4_pollfd_struct_t poll_fd = {};
-	poll_fd.fd = _local_pos_sub;
-	poll_fd.events = POLLIN;
-
 	while (!should_exit()) {
-		// poll for new data on the local position state topic (wait for up to 20ms)
-		const int poll_return = px4_poll(&poll_fd, 1, 20);
-
-		// poll_return == 0: go through the loop anyway to copy manual input at 50 Hz
-		// this is undesirable but not much we can do
-		if (poll_return < 0) {
-			PX4_ERR("poll error %d %d", poll_return, errno);
+		// wait for new data on the local position state topic
+		if (!_local_pos_sub.updateBlocking(_local_pos)) {
```

Comment:
With the previous implementation we still ran this loop on timeout, with the new one we're just continuing, right? Do we know the implications?

Reply:
@MaEtUgR could you check?

Reply:
This is solved in a different way now right? https://github.com/PX4/Firmware/blob/f956bafa4eda7ceb298790ef0fff5540fd3c589f/src/modules/mc_pos_control/mc_pos_control_main.cpp#L535
I think it's much better to run the controller up on new estimation data but we need to make sure there's instant and reliable emergency handling in case of a timeout. Last resort the attitude controller should keep the vehicle stable level if attitude setpoints are missing.

Reply:
Yes it will need to be updated, although that entire topic needs to be reviewed. Commander and mc_pos_control have separate notions of position and velocity validity.

Reply:
Actually, mc_pos_control was pulled out of this PR (moved to a WQ).